### PR TITLE
freecad: update to 0.21.1.

### PR DESCRIPTION
**Pull request by cy1der**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture: **x86_64-glibc**
- I built this PR locally for these architectures: **Only tried armv6l, but I got "hdf5-devel cannot be cross compiled"**

### CODE REVIEW

**Comment by ii8** [comment]
You should squash the commits and the commit message should be: "freecad: update to 0.21.1."
Have a look here if you don't know how: https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history
Why does it need the webengine stuff now? Just curious.

**Comment by cy1der** [comment]
> Why does it need the webengine stuff now? Just curious.
Unsure, but my guess is it's for the built in web browsing capabilities.

**Comment by cy1der** [comment]
> You should squash the commits and the commit message should be: "freecad: update to 0.21.1."

Sorry about that, will do

**Comment by ii8** [comment]
I didn't mean the pull request title. The pull request title doesn't matter so much. I meant the commit message.
You already squashed it correctly so all you have to do to fix the message is:
```
git commit --amend -m 'freecad: update to 0.21.1.'
```

**Comment by robertek** [comment]
To be noted when updating the FreeCAD that it would be good to include the `python3-GitPython python3-Markdown` as runtime deps. They are needed for the Addon manager to show correctly the addons.

See my branch:
https://github.com/void-linux/void-packages/compare/master...robertek:void-packages:freecad

**Comment by cy1der** [comment]
Done, thanks for the suggestion

**Comment by zlice** [comment]
anyone else get python 3.12 issues? not sure if those need patched or what

the first ones are mods i pulled, but the bottom ones are from the install

```
  #####                 ####  ###   ####  
  #                    #      # #   #   # 
  #     ##  #### ####  #     #   #  #   # 
  ####  # # #  # #  #  #     #####  #   # 
  #     #   #### ####  #    #     # #   # 
  #     #   #    #     #    #     # #   #  ##  ##  ##
  #     #   #### ####   ### #     # ####   ##  ##  ##

<string>:439: SyntaxWarning: invalid escape sequence '\s'
During initialization the error "No module named 'imp'" occurred in /home/zlice/.local/share/FreeCAD/Mod/kicadStepUpMod/./InitGui.py
Please look into the log file for further information
<string>:172: SyntaxWarning: invalid escape sequence '\s'
During initialization the error "No module named 'imp'" occurred in /home/zlice/.local/share/FreeCAD/Mod/Manipulator/./InitGui.py
Please look into the log file for further information
The `.FCStd#` backup format is deprecated as of v0.21 and may be removed in future versions.
To update, check the 'Preferences->General->Document->Use date and FCBak extension' option.
Traceback (most recent call last):
  File "/usr/lib/freecad/Mod/Draft/draftviewproviders/view_base.py", line 179, in attach
    self.onChanged(vobj, "Pattern")
  File "/usr/lib/freecad/Mod/Draft/draftviewproviders/view_base.py", line 286, in onChanged
    if str(vobj.Pattern) in list(utils.svg_patterns().keys()):
                                 ^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/freecad/Mod/Draft/draftutils/utils.py", line 789, in svg_patterns
    loadSvgPatterns()
  File "/usr/lib/freecad/Mod/Draft/draftutils/utils.py", line 763, in load_svg_patterns
    p = importSVG.getContents(file, 'pattern')
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/freecad/Mod/Draft/importSVG.py", line 1705, in getContents
    f = pythonopen(filename)
        ^^^^^^^^^^
<class 'NameError'>: name 'pythonopen' is not defined
```

**Comment by cy1der** [comment]
Python 3.12 works just fine for me, but I don't have the "Manipulator" and "kicadStepUpMod".

Have you tried disabling or removing them?

File: srcpkgs/freecad/template [hunk 1]
```diff
@@ -24,8 +24,9 @@ makedepends="python3-devel boost-devel libxerces-c-devel zlib-devel occt-devel
  coin3-devel libshiboken2-devel libspnav-devel pyside2-tools
  liblz4-devel libpyside2-python3-devel python3-matplotlib netcdf-devel
  jsoncpp-devel qt5-devel qt5-svg-devel qt5-tools-devel qt5-webkit-devel
- qt5-xmlpatterns-devel coin3-doc glew-devel python3-pivy"
-depends="python3-matplotlib python3-pyside2-webengine python3-pivy"
+ qt5-xmlpatterns-devel coin3-doc glew-devel python3-pivy qt5-webengine-devel
+ qt5-declarative-devel qt5-webchannel-devel qt5-location-devel"
+depends="python3-matplotlib python3-pyside2-webengine python3-pivy python3-GitPython python3-Markdown"
```
Comment (ahesford) [inline srcpkgs/freecad/template:29]: Wrap lines at 80 columns.

**Review by ahesford — DISMISSED** [review]

**Comment by zlice** [comment]
Does look like something to fix with the Mods. Moving my mod dir gets rid of all the python errors.

**Comment by cy1der** [comment]
Anything I need to do to the template or nah?

**Comment by zlice** [comment]
Actually looking at it again, just fixing the Manipulator mod isn't enough.

I think the FreeCAD Draft/importSVG mod needs patched ~1705

```
--- ./src/Mod/Draft/importSVG.py	2023-08-30 22:32:05.000000000 -0400
+++ /lib/freecad/Mod/Draft/importSVG.py	2023-12-09 11:41:56.315252658 -0500
@@ -1702,7 +1702,10 @@
         contents = filename
     else:
         # Use the native Python open which was saved as `pythonopen`
-        f = pythonopen(filename)
+        if (sys.version_info < (3, 12)):  #py3
+          f = pythonopen(filename)
+        else:
+          f = open(filename)
         contents = f.read()
         f.close()
```

**Pull request closed without merging**